CMakeLists.txt

```diff
@@ -34,6 +34,33 @@ if (MSVC)
   set(CMAKE_CXX_STANDARD_REQUIRED ON)
 endif()
 
+set(COMPILER ${CMAKE_CXX_COMPILER_ID})
+if (COMPILER STREQUAL "GNU")
+  set(COMPILER "GCC") # perfer printing GCC instead of GNU
+endif()
+if (COMPILER MATCHES "Clang")
+  set(COMPILER "Clang") # Simplify all Clang variants to just Clang
+endif()
+
+# Enforce minimium compiler versions that support the c++20 features we use
+set (GCC_min_version 10)
+set (Clang_min_version 11)
```

Comment:
[As I noted here](https://github.com/dolphin-emu/dolphin/pull/10969#issuecomment-1435946039), Dolphin ends up requiring a newer Clang than this, due to https://github.com/dolphin-emu/dolphin/commit/154cb4f722a9996a4134463e64c85c6bfa25d8bf. See "Class types in non-type template parameters" in https://en.cppreference.com/w/cpp/compiler_support.

Reply:
Ok, I've bumped it to Clang 12 (Apple Clang 13)